src/main/java/meditracker/medication/MedicationManager.java

```diff
@@ -72,6 +77,30 @@ private static void checkForDuplicateMedication(String name) throws MediTrackerE
         }
     }
 
+    /**
+     * Checks if the user has input a valid expiry date format.
+     * Also checks that the expiry date entered is not expired.
+     *
+     * @param expiryDate Expiry Date in yyyy-MM-dd format
+     * @throws DateTimeParseException When the date entered is in the wrong format
+     * @throws MediTrackerException When the date entered is already expired
+     */
+    public static void checkforValidExpiryDate(String expiryDate)
+            throws DateTimeParseException, MediTrackerException {
+        try {
+            DateTimeFormatter dateTimeFormatter = DateTimeFormatter.ofPattern("yyyy-MM-dd");
+            LocalDate parsedExpiryDate = LocalDate.parse(expiryDate, dateTimeFormatter);
+            LocalDate currentDate = MediTrackerTime.getCurrentDate();
+
+            if (parsedExpiryDate.isBefore(currentDate)) {
+                throw new MediTrackerException("You are not allowed to enter expired medications!");
+            }
+
+        } catch (DateTimeParseException e) {
+            throw new MediTrackerException("Please enter a valid expiry date in yyyy-MM-dd!");
+        }
```

Comment:
Might be better to scope the try catch to the line(s) that will throw the exception

```suggestion
        DateTimeFormatter dateTimeFormatter = DateTimeFormatter.ofPattern("yyyy-MM-dd");
        LocalDate parsedExpiryDate;
        try {
            parsedExpiryDate = LocalDate.parse(expiryDate, dateTimeFormatter);
        } catch (DateTimeParseException e) {
            throw new MediTrackerException("Please enter a valid expiry date in yyyy-MM-dd!");
        }
        LocalDate currentDate = MediTrackerTime.getCurrentDate();
        if (parsedExpiryDate.isBefore(currentDate)) {
            throw new MediTrackerException("You are not allowed to enter expired medications!");
        }
```

Reply:
Implemented in [03dcdc0](https://github.com/AY2324S2-CS2113T-T09-1/tp/pull/220/commits/03dcdc0cbbfea652ba737bdb9ce7952053dd7d18)